src/models/transaction.ts

```diff
@@ -37,12 +39,17 @@ export type Transaction = IndexerTransaction & {
   parsed: ParsedTransactionData;
 };
 
+type AccountsHash = string;
+type BlockNumber = 'latest' | `${number}`;
+type CacheKey = `${AccountsHash}:${BlockNumber}`;
+
 class TransactionStore {
   realmSchemaName = TransactionRealmObject.schema.name;
   private _transactions: Array<Transaction> = [];
-  private _lastSyncedAccountsHash = '';
-  private _lastSyncedBlockNumber = 'latest';
+  private _lastSyncedAccountsHash: AccountsHash = '';
+  private _lastSyncedBlockNumber: BlockNumber = 'latest';
   private _isLoading = false;
+  private _cache = new Map<CacheKey, Transaction[]>();
```

Comment:
Is it working ok with `makeAutoObservable`?

Reply:
yep